beetsplug/bandcamp/http.py

```diff
@@ -0,0 +1,34 @@
+import http.client
+from functools import lru_cache
+from html import unescape
+from urllib.parse import urlsplit
+
+from beets import __version__
+
+
+USER_AGENT = f"beets/{__version__} +https://beets.io/"
+
+
+class HTTPBadStatus(http.client.HTTPException):
+    """HTTP request ended with non OK status."""
+
+    def __init__(self, url, response):
+        super().__init__(f"HTTP GET to {url} failed: {response.status} {response.reason}")
+        self.url = url
+        self.response = response
+
+
+@lru_cache(maxsize=None)
+def http_get_text(url: str) -> str:
+    """Return text contents of the url."""
+
+    host = urlsplit(url).hostname
+    conn = http.client.HTTPSConnection(host)
```

Comment:
While your implementation here fixes the problem and is actually more straightforward than what I would expect a replacement to `requests.get` to look like, it adds a fair bit of complexity.

Have you tried using `httpx` to see whether it fails in the same way?

Reply:
Yeah it doesn't reproduce the whole functionality (for instance I don't think this respect's proxy env vars).  
I'll try a couple more combinations, and look at using a "session" for TCP connection re-use, which should be pretty nice since we're mostly connecting to a single domain!

Reply:
I like the idea of connection pooling, check out `httpx.Client` which has the same functionality as `requests.Session`. Making it a singleton should be as simple as initialising it at the top of this module once it's being imported.

Reply:
`httpx` also works for me, I guess it's another data point for urllib3 being the issue.
I amended the last commit.